### DISCUSSION TITLE
8월 21일 중간점검 발표내용

### DISCUSSION THREAD

**Discussion by aicompose** [CodeMedics/Harulog-backend#50] [Ideas]
# 8월 21일 중간점검

---
- 테스트 코드, 이슈 + 깃허브 기반
- c, u, d 기존 서비스 로직을 작성한 것으로 기반으로 테스트코드
- r TDD 기반으로 
- 리액트 화면 구상
- 앞으로의 계획 설명하기

**Comment by ej20001000** [comment]
앞으로 남은 스케줄 동안 언제까지 무엇을 할 것인지
깃 브렌치 전략 설명
Test 코드를 어떻게 짤지에 대한 고민을 설명

**Comment by ej20001000** [comment]
정리사항

- 백승진: 테스트 코드 작성에 대한 고민
- 조평훈: 앞으로 스케줄에 대한 대략적 로드맵

TDD 기반 개발 어떻게 할건지, 리액트 컴포넌트 구성

**Comment by ej20001000** [comment]
TDD 기반 테스트를 어떻게 할 것인가?

1. Read 관련 기능들에 대한 결과 값을 도출한다
 1-1. 피그마 화면 구성을 보며 결과 값을 도출하였다
ex) 댓글> 댓글 내용 닉네임 날짜 등등

2. 이에 기반 하여 Assert 코드 작성

**Reply by ej20001000** [reply]
AssertNotNull - 엔티티
AssertEquals - 엔티티 내용 확인

**Reply by aicompose** [reply]
ㅇㅋ

**Reply by ej20001000** [reply]
ok

**Comment by ej20001000** [comment]
package com.ohgiraffers.dailylogbackend.diary.update;

import com.ohgiraffers.dailylogbackend.diary.command.application.dto.DiaryUpdateDTO;
import com.ohgiraffers.dailylogbackend.diary.command.application.service.DiaryServiceImpl;
import com.ohgiraffers.dailylogbackend.diary.command.domain.aggregate.entity.DiaryEntity;
import com.ohgiraffers.dailylogbackend.diary.command.infra.repository.DiaryRepository;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

@SpringBootTest
public class DiaryUpdateTest {

    @Autowired
    private DiaryServiceImpl diaryServiceImpl;

    @Autowired
    private DiaryRepository diaryRepository;

    @Test
    @DisplayName("diary update test")
    public void diaryUpdateTest() {
        // Arrange

        DiaryUpdateDTO diaryUpdateDTO = new DiaryUpdateDTO("meme", "ojingoo");

        // Act
        DiaryEntity updatedDiary = diaryServiceImpl.updateDiary(8L, diaryUpdateDTO); // serviceImpl 안의 save 코드가 저장된 엔티티 값을 리턴함

        // Assert
        assertNotNull(updatedDiary); // 리턴된 엔티티가 null 값이 아닌지 확인
        assertEquals(diaryUpdateDTO.getDiaryContent(), updatedDiary.getDiaryContent()); // 리턴된 엔티티 안의 값이 처음 지정한 값과 일치하는지 확인
        assertEquals(diaryUpdateDTO.getFeelCategory(), updatedDiary.getFeelCategory());
    }
}

**Comment by Oleevia01** [comment]
1주차 : 프로젝트 컨셉 논의 및 정리, 피그마&미로 설계, Entity 구상 및 작성
2주차 : 기본적인 C, U, D 기능 구현 및 테스트 코드 작성
3주차 : 리액트 컴포넌트 및 기본 틀 구성

**Comment by byallfine01** [comment]
[리액트 컴포넌트 및 기본 틀 구성]

commons(공통 컴포넌트 관리 폴더) :  피드, 내 일기장, 마이페이지, 피드백, 로그인, 로그아웃으로 연결되는,
왼쪽 레이아웃 bar 로,
모든페이지 동일하게 사이드바구성 및 가운데 위에 harulog 로고 구성(Header, Footer, CommonSideBar, LogoSideBar)

modules : api feature 단위 관리 - 카카오 로그인 api,  각 페이지 별로 api 호출 url 설정 및 함수 호출

pages  : 피드, 내 일기장, 마이페이지, 피드백, 관리자 페이지별로  컴포넌트 구성

css :  피드, 내 일기장, 마이페이지, 피드백, 관리자 페이지 별로 css 파일 각각 구성

** 전체적으로 로그인부터 모든 페이지 사이즈 동일하게 레이아웃 구성,

** 내 일기장 페이지에서 왼쪽엔 달력 컴포넌트, 오른쪽엔 "일주일간 나의 기분"의 통계 컴포넌트 구성

**Comment by byallfine01** [comment]
### 8/27(일) 까지 일정 

- 로그인 (조평훈)
   ~8/21(월) 백엔드 마무리, 이후 화면 설계 예정


- 일기 (백승진)
  ~ 8/27일 댓글, 파일, 신고와 연계하여 react 표출

- 파일 (정민호)
  ~ 8/21(월) 기초 C, U, D 테스트 코드 완료
  8/22(화)~8/27(일) 화면 설계 react 적용

- 댓글 (이주원)
  ~ 8/21(월) 기초 C, U, D 테스트 코드 완료
  8/22(화) ~ 8/27(일) 화면 설계  

- 신고 (남효정)
  ~ 8/20(일) 기초 C, U, D 테스트 코드,
  ~ 8/23(수) controller 및 신고 정책 구현, 이후 화면 설계 예정
